### PR TITLE
[Feature] Update native OneSignal SDKs dependencies to 5.1.x

### PR DESCRIPTION
**Pull request by shepherd-l**
# Description
## One Line Summary
Updated included OneSignal Android SDK to [5.1.1](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.1) and OneSignal iOS SDK to [5.1.0](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.1.0)
### Motivation
Applies native fixes from the OneSignal Android SDK and OneSignal iOS SDK to the Unity SDK
- [OneSignal Android SDK 5.1.1 Release Notes](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.1)
  - Includes location behaviour change from [5.1.0](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.0):
     - You will need to include location permission in your own app's manifest
- [OneSignal iOS SDK 5.1.0 Release Notes](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/5.1.0)

### Scope
Unity developers must target API level 33 or higher to build for Android. This is because androidx.work:work-runtime-ktx:2.8.1 from the [OneSignal Android SDK](https://github.com/OneSignal/OneSignal-Android-SDK/pull/1960) requires compileSdkVersion 33 or higher.
Older versions of Unity do not include Android 33 by default - 2022 and below, so users will have to download it. Steps on how to do so are listed in the Unity docs: [Setting the Android SDK Target API](https://docs.unity3d.com/2023.3/Documentation/Manual/android-sdksetup.html).

# Testing
## Manual testing
Tested OneSignal initialization, app build with Unity 2022.3.10f1 of the OneSignal example app on a emulated Pixel 4 with Android 12 and physical iPhone 12 with iOS 15.5.

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [ ] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/689)

### CODE REVIEW

**Comment by jinliu9508** [comment]
@shepherd-l Is there any reason that we update IOS to 5.1.0 instead of 5.1.1?

**Comment by shepherd-l** [comment]
@jinliu9508 I think 5.1.0 is the latest release of iOS. Could you show me where you found a 5.1.1 release?

**Comment by nan-li** [comment]
This sounds ok to me, since you must target Android 33 to release to the Play Store now, anyway.

**Comment by jinliu9508** [comment]
> @jinliu9508 I think 5.1.0 is the latest release of iOS. Could you show me where you found a 5.1.1 release?

Yes you are correct, 5.1.1 IOS is only in draft mode. This looks good to me know.

File: com.onesignal.unity.android/Editor/OneSignalAndroidDependencies.xml [hunk 1]
```diff
@@ -3,6 +3,6 @@
     <repositories>
       <repository>https://repo.maven.apache.org/maven2</repository>
     </repositories>
-    <androidPackage spec="com.onesignal:OneSignal:5.0.5" />
+    <androidPackage spec="com.onesignal:OneSignal:5.1.1" />
```
[the remark this answers is not in the document]
Comment (jinliu9508) [reply com.onesignal.unity.android/Editor/OneSignalAndroidDependencies.xml:6]: This is due to targeting api level 33 and higher. @shepherd-l has updated the documentation to reflect this. Looks good to me now.

**Review by jinliu9508 — APPROVED** [review]

**Pull request merged**